src/vehicle.cpp

```diff
@@ -3752,9 +3762,59 @@ void vehicle::idle(bool on_map) {
 
     if( on_map ) {
         update_time();
+        if(scoop_on){
+            operate_scoop();
+        }
+    }
+}
+void vehicle::operate_scoop()
+{
+    std::vector<int> scoops = all_parts_with_feature( "SCOOP" );
+    auto veh_points = get_points();
+    for( int scoop : scoops ) {
+        const int chance_to_damage_item = 9;
+        int max_pickup_size = parts[scoop].info().size/10;
+        const char *sound_msgs[] = {_("Whirrrr"), _("Ker-chunk"), _("Swish"), _("Cugugugugug")};
+        sounds::sound( global_pos3() + parts[scoop].precalc[0], rng( 20, 35 ), sound_msgs[rng( 0, 3 )] );
+        std::vector<tripoint> part_points;
+        for(int x = -1;x <= 1;x++){
```

Comment:
Much cleaner way to do it would be:
```
for( const tripoint &current : g->m.points_in_radius( global_pos3() + parts[scoop].precalc[0], 1 ) ) {
    part_points.push_back(current);
}
```
This will also require a `#include "map_iterator.h"` somewhere near the top of the file.